### PR TITLE
Fix/subscription payment mode column

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

When the `payment_mode` column was added to the subscription table, we missed including it in the insert/update methods on the subscription repository so that the new column was not saved once the model was saved.

## Affects

- `SubscriptionRepository->insert()` and `SubscriptionRepository->update()` methods now have the `mode` property to be inserted or updated to the database;
- The `SubscriptionFactory` now brings a value for the `mode` property;
- The `mode` parameter has been added to the mocked request.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@pauloiankoski can you please refer me to the PR or when the subscription model added a `mode` property?  Some of these sub-epic PR's are getting kind of hard to follow the trail for context.

**Comment by pauloiankoski** [comment]
> @pauloiankoski can you please refer me to the PR or when the subscription model added a `mode` property? Some of these sub-epic PR's are getting kind of hard to follow the trail for context.

@jonwaldstein It was added specifically on [this commit](https://github.com/impress-org/givewp/pull/6616/commits/ac009d59f34af2aa330732e613dea2e27da9eba9) that was merged within [this PR](https://github.com/impress-org/givewp/pull/6616)

**Review by jonwaldstein — APPROVED** [review]
👍
